server/aws-lsp-codewhisperer/src/language-server/netTransform/transformHandler.ts

```diff
@@ -289,10 +290,17 @@ export class TransformHandler {
                     break
                 }
             } catch (e: any) {
-                const errorMessage = (e as Error).message ?? 'Error in GetTransformation API call'
-                this.logging.log('CodeTransformation: GetTransformation error = ' + errorMessage)
-                status = 'FAILED'
-                break
+                if (status === 'Retry' && getTransformRetry === getTransformMaxRetry) {
+                    const errorMessage = (e as Error).message ?? 'Error in GetTransformation API call'
+                    this.logging.log('CodeTransformation: GetTransformation error = ' + errorMessage)
+                    status = 'FAILED'
+                    break
+                } else if (status === 'Retry') {
+                    getTransformRetry += 1
+                } else {
+                    status = 'Retry' // first time failed, status will be overwritten if no failure in next poll
+                    getTransformRetry = 0
```

Comment:
Imagine the experience as the end user. Would it be helpful to see the error and that the system is retrying? Would the current attempt and max attempt count also help?